creativecloud/blocks/interactive-marquee/interactive-marquee.js

```diff
@@ -0,0 +1,77 @@
+import { setLibs } from '../../scripts/utils.js';
+
+const miloLibs = setLibs('/libs');
+
+const { decorateButtons, decorateBlockBg } = await import(`${miloLibs}/utils/decorate.js`);
+const { createTag, loadStyle } = await import(`${miloLibs}/utils/utils.js`);
+
+// [headingSize, bodySize, detailSize, titlesize]
+const typeSizes = ['xxl', 'xl', 'l', 'xs'];
+
+function decorateText(el) {
+  const headings = el.querySelectorAll('h1, h2, h3, h4, h5, h6');
+  const heading = headings[headings.length - 1];
+  const config = typeSizes;
+  const decorate = (headingEl, typeSize) => {
+    headingEl.classList.add(`heading-${typeSize[0]}`);
+    const bodyEl = headingEl.nextElementSibling;
+    bodyEl?.classList.add(`body-${typeSize[1]}`);
+    bodyEl?.nextElementSibling?.classList.add(`body-${typeSize[1]}`, 'pricing');
+    const sib = headingEl.previousElementSibling;
+    if (sib) {
+      const className = sib.querySelector('img, .icon') ? 'icon-area' : `detail-${typeSize[2]}`;
+      sib.classList.add(className);
+      sib.previousElementSibling?.classList.add('icon-area');
+    }
+    const iconAreaElements = el.querySelector('.icon-area');
+    const iconText = createTag('div', { class: `heading-${typeSize[3]} icon-text` });
+    iconAreaElements.appendChild(iconText);
+    iconAreaElements?.classList.add('icon-area');
+    iconText.innerText = (iconAreaElements.textContent.trim());
+    iconText.previousSibling.textContent = '';
+  };
+  decorate(heading, config);
+}
+
+function extendButtonsClass(text) {
+  const buttons = text.querySelectorAll('.con-button');
+  if (buttons.length === 0) return;
+  buttons.forEach((button) => { button.classList.add('button-justified-mobile'); });
+}
+
+function interactiveInit(el) {
```

Comment:
can we make this function async? I have an await call to add in this function to load firefly js file.